python/tests/kernel/test_kernel_features.py

```diff
@@ -1310,3 +1316,93 @@ def test():
     
     counts = cudaq.sample(test)
     assert len(counts) == 1 and '101' in counts
+
+
+
+def test_bad_return_value_with_stdvec_arg():
+
+    @cudaq.kernel
+    def test_param(i: int, l: List[int]) -> int:
+        return i
+
+    l = [42]
+    for i in range(4):
+        assert test_param(i, l) == i
+
+
+def test_u3_op():
+
+    @cudaq.kernel
+    def check_x():
+        q = cudaq.qubit()
+        # implement Pauli-X gate with U3
+        u3(np.pi, np.pi, np.pi / 2, q)
+
+    print(check_x)
+    counts = cudaq.sample(check_x)
+    assert counts["1"] == 1000
+
+    @cudaq.kernel
+    def bell_pair():
+        qubits = cudaq.qvector(2)
+        # implement Hadamard gate with U3
+        u3(np.pi / 2, 0, np.pi, qubits[0])
+        cx(qubits[0], qubits[1])
+
+    counts = cudaq.sample(bell_pair)
+    assert (len(counts) == 2)
+    assert ('00' in counts)
+    assert ('11' in counts)
+
+
+def test_u3_ctrl():
+
+    @cudaq.kernel
+    def another_bell_pair():
+        qubits = cudaq.qvector(2)
+        u3(np.pi / 2, 0, np.pi, qubits[0])
+        u3.ctrl(np.pi, np.pi, np.pi / 2, qubits[0], qubits[1])
+
+    print(another_bell_pair)
+    counts = cudaq.sample(another_bell_pair)
+    assert (len(counts) == 2)
+    assert ('00' in counts)
+    assert ('11' in counts)
+
+
+def test_u3_adj():
+
+    @cudaq.kernel
+    def rotation_adjoint_test():
+        q = cudaq.qubit()
+
+        # # implement Rx gate with U3
+        # u3(1.1, -np.pi / 2, np.pi / 2, q)
```

Comment:
@boschmitt : Can you please check the Rx representation?

Reply:
Fixed in [20c9b87](https://github.com/NVIDIA/cuda-quantum/pull/1504/commits/20c9b87086e68f18af7438356ad22a71f70502b7)